bezier_curve_es.frag is broken in wasm

### Checklist

- [x] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [x] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

Running any fyne application in the wasm environment results in the following crash:
```
panic: failed to compile OpenGL shader: [wasm_exec.js:22:14](http://localhost:8080/wasm_exec.js)
	ERROR: 0:102: 'i' : Loop index cannot be compared with non-constant expression [wasm_exec.js:22:14](http://localhost:8080/wasm_exec.js)
	[wasm_exec.js:22:14](http://localhost:8080/wasm_exec.js)
	>>> SHADER SOURCE [wasm_exec.js:22:14](http://localhost:8080/wasm_exec.js)
	#version 100 [wasm_exec.js:22:14](http://localhost:8080/wasm_exec.js)
	[wasm_exec.js:22:14](http://localhost:8080/wasm_exec.js)
	#ifdef GL_ES [wasm_exec.js:22:14](http://localhost:8080/wasm_exec.js)
	# ifdef GL_FRAGMENT_PRECISION_HIGH [wasm_exec.js:22:14](http://localhost:8080/wasm_exec.js)
	precision highp float; [wasm_exec.js:22:14](http://localhost:8080/wasm_exec.js)
	# else [wasm_exec.js:22:14](http://localhost:8080/wasm_exec.js)
	precision mediump float; [wasm_exec.js:22:14](http://localhost:8080/wasm_exec.js)
	#endif [wasm_exec.js:22:14](http://localhost:8080/wasm_exec.js)
	precision mediump int; [wasm_exec.js:22:14](http://localhost:8080/wasm_exec.js)
	precision lowp sampler2D; [wasm_exec.js:22:14](http://localhost:8080/wasm_exec.js)
	#endif
       ...snip
```

### How to reproduce

1. Run a fyne application with `fyne serve`
2. open the console in a browser window, and navigate to localhost:8080
3. Observe crash written to console.

### Screenshots

_No response_

### Example code

```go
package main

import (
        "fyne.io/fyne/v2/app"
        "fyne.io/fyne/v2/widget"
)

func main() {
        a := app.NewWithID("test")
        w := a.NewWindow("Test")

        w.SetContent(widget.NewLabel("Hello, world"))

        w.ShowAndRun()
}
```

### Fyne version

develop

### Go compiler version

go1.25.3

### Operating system and version

Gentoo

### Additional Information

_No response_

--- a/internal/painter/gl/shaders/bezier_curve_es.frag
+++ b/internal/painter/gl/shaders/bezier_curve_es.frag
@@ -85,9 +85,10 @@ float quadratic_distance(vec2 pos, vec2 v0, vec2 v1, vec2 v2)
 float cubic_distance(vec2 p, vec2 v0, vec2 v1, vec2 v2, vec2 v3)
 {
     const int NEWTON_ITERS = 4;
+    const int MAX_STEPS = 64;
 
     float approx_length = length(v1 - v0) + length(v2 - v1) + length(v3 - v2); // control polygon length
-    int steps = int(clamp(approx_length * 10.0, 16.0, 64.0)); // scale steps by length
+    int steps = int(clamp(approx_length * 10.0, 16.0, float(MAX_STEPS))); // scale steps by length
 
     vec2 a = -v0 + 3.0*v1 - 3.0*v2 + v3;
     vec2 b =  3.0*v0 - 6.0*v1 + 3.0*v2;
@@ -99,8 +100,10 @@ float cubic_distance(vec2 p, vec2 v0, vec2 v1, vec2 v2, vec2 v3)
     float best_seg_dist = 1e3;
     float t0 = 0.0;
 
-    for (int i = 1; i <= steps; i++)
+    for (int i = 1; i <= MAX_STEPS; i++)
     {
+        if (i > steps) break;
+
         float t = float(i) * dt;
         vec2 cur = ((a*t + b)*t + c)*t + d;
 

